automations/vision_filter.py

```diff
@@ -104,8 +91,8 @@ def update(self):
     def execute(self):
         if self.vision.time == 0:
             return
-        vision_delay = self.vision.dt + self.control_loop_average_delay + (time.time() - self.last_vision_local_time)
-        timesteps_since_vision = int(vision_delay/50)
+        vision_delay = self.vision.dt + self.control_loop_average_delay
+        timesteps_since_vision = int(vision_delay*50)
```

Comment:
Cause the code was hastily written at the regional in one of the huge patch bombs and not really tested or reviewed properly. I only caught it when I realised it wasn't predicting forward in simulation. I think in future when writing things like this with so many moving parts we should really try to simulate as much as possible to avoid errors like this.